### PR TITLE
Fix scrollbars in public template

### LINKED ISSUE #18968: Design of files_sharing's public template needs to be upgraded
### Private files
private_files

### Public files
public_files

### Notes 

Fixed in Gallery, but the structure in Files is not the same... so the files_sharing experts should step in.
https://github.com/owncloud/gallery/commit/d7bdfab556b3a6500ce16e61ff9db4d1a49c4e99

@jancborchardt

### PR DESCRIPTION
**Pull request by Henni**
fixes #18968

@oparoz

### CODE REVIEW

**Comment by oparoz** [comment]
@Henni Just to be sure, this only takes care of the scrollbar and the footer will be addressed separately or are you trying to get both to work at the same time?

I'm mentioning it because there is a huge gap between the last file and the footer when it's not empty.

**Comment by oparoz** [comment]
Ah, the same gap is found on master... so, not related to this PR.

**Comment by Henni** [comment]
This gap also exists in the filelist in the files app.
I could remove it. I don't even know why it is there in the files app.

**Comment by oparoz** [comment]
> I could remove it. I don't even know why it is there in the files app.

Most of it was for the share dialogue, instead of pushing the element into view. See https://github.com/owncloud/core/pull/15805

**Comment by Henni** [comment]
ok that makes sense.

**Comment by oparoz** [comment]
OK, so that gap can be removed in another PR since we don't need that space any more.

In this PR though, there is a difference between an empty page on the private and on the public side.
On the private side, the message starts at 30%, but on the public side it's glued to the controls. I had the same problem on Gallery. It's because the footer is relative positioned and the empty message absolute.

**Comment by MorrisJobke** [comment]
Easier to review: https://github.com/owncloud/core/pull/19611/files?w=1

**Comment by Henni** [comment]
Empty message now looks like this:
image

**Comment by oparoz** [comment]
That's a good fix and actually a revert of https://github.com/owncloud/core/pull/16469 which was put in place to fix the old message, so unneeded for 8.2+

There is one small problem when adding content to the folder: the footer overlaps the last row in Gallery. Hopefully I can fix this with some margin-top

**Comment by Henni** [comment]
@oparoz Let me fix this.
Should the gallery release for 8.2 work with owncloud 8.1?

**Comment by Henni** [comment]
image

@oparoz maybe you are able to remove the scrollbar in the gallery view.

**Comment by oparoz** [comment]
> Should the gallery release for 8.2 work with owncloud 8.1?

It would be nice since Gallery+ is for older versions of ownCloud

> maybe you are able to remove the scrollbar in the gallery view

Not sure, but I can try :)

**Comment by oparoz** [comment]
@Henni What do you think about the fact that the footer appears just under the empty message and not at the bottom of the page?
On a page with content, it appears just below the last row because I think it would involve some JS to put it at the bottom and get it pushed by rows, but on an empty page, at least until the apps are capable of automatically fill the screen as files are uploaded, there is no need to move it and maybe the footer should be at the bottom?

**Comment by Henni** [comment]
image

**Comment by MorrisJobke** [comment]
Tested and works :+1:

**Comment by MorrisJobke** [comment]
@owncloud/designers @PVince81 Please review

**Comment by MorrisJobke** [comment]
@cmonteroluque Maybe include this into 8.2 because it fixes https://github.com/owncloud/gallery/issues/374

**Comment by oparoz** [comment]
There is a regression. There is now a double scrollbar when adding content to the folder.

**Comment by Henni** [comment]
fixed regression

**Comment by oparoz** [comment]
I think it now works well.

Tested on Firefox

## Files

### Empty page
- [x] No scrollbar
- [x] Footer at the bottom
- [x] Empty message at 30%

### Page with only a couple of files

- [x] No scrollbar
- [x] Footer at the bottom of the row.

### Page with 3/4 content

- [ ] No scrollbar
- [ ] Footer at the bottom of the row. It's currently too far and the reason there is a scrollbar. TBF in separate PR

### Page with list longer than the page

- [x] No double scrollbar
- [ ] Footer at the bottom of the row

## Gallery

### Empty page
- [x] No scrollbar (with https://github.com/owncloud/gallery/pull/414)
- [ ] Footer at the bottom
- [x] Empty message at 30%

### Page with only a couple of files

- [x] No scrollbar (with https://github.com/owncloud/gallery/pull/414)
- [x] Footer at the bottom of the row.

### Page with 3/4 content

- [x] No scrollbar (with https://github.com/owncloud/gallery/pull/414)
- [x] Footer at the bottom of the row.

### Page with list longer than the page

- [x] No double scrollbar
- [x] Footer at the bottom of the row

**Comment by MorrisJobke** [comment]
I tested following on Firefox, Chromium and IE8:

* [x] single file share
* [x] empty folder share
* [x] folder share with less than screen size elements
* [x] folder share with more than screen size elements

All works well :+1:

Thanks @Henni and @oparoz

**Comment by MorrisJobke** [comment]
@DeepDiver1975 @cmonteroluque We might want this in 8.2 ;) Feel free to merge if you are okay with this

**Comment by MorrisJobke** [comment]
Ah and I forgot:

* [x] switch to gallery app with no images
* [x] switch to gallery app with images

Both worked well and the slideshow also started.

**Comment by oparoz** [comment]
@MTRichards , please confirm that this is good enough for IE8. Works for me ;)

ie8

**Comment by oparoz** [comment]
Regarding the switch to Gallery, there is a regression in Gallery, the button to switch to Files on an empty page is missing...

**Comment by MTRichards** [comment]
>@MTRichards , please confirm that this is good enough for IE8. Works for me ;)

There are a bunch of checkboxes above still open, most seem minor. While in a perfect world it should be perfect, IE 8 is hardly a perfect world. From what I can tell, it is usable, and a few minor cosmetic issues seem to be a problem. ok for me at this time.

**Comment by oparoz** [comment]
Thanks

**Comment by cmonteroluque** [comment]
https://github.com/owncloud/core/pull/19611#issuecomment-146152996 a good thing :)

File: core/css/styles.css [hunk 1]
```diff
@@ -425,9 +424,9 @@ input[type="submit"].enabled {
 .emptycontent {
 	font-size: 16px;
 	color: #888;
-	position: absolute;
 	text-align: center;
-	top: 30%;
+	margin-top: 100px; /* ie8 */
+	margin-top: 30vh;
```
Comment (nickvergessen) [inline core/css/styles.css]: why does this global class suddenly receive a margin-top so close to a release.
Now you broke `.emptycontent` for all apps that use it without fullpage-size, e.g. notifications and announcementcenter

**Comment by PVince81** [comment]
Regression: https://github.com/owncloud/core/issues/19886

**Comment by LukasReschke** [comment]
Regression: https://github.com/owncloud/core/issues/20557

**Pull request merged**